presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PushDownTableConstraints.java

```diff
@@ -143,7 +145,7 @@ private boolean planChanged(PlanNode rewrittenPlan, FilterNode oldPlan, Lookup l
         }
 
         FilterNode rewrittenFilter = (FilterNode) rewrittenPlan;
-        if (!rewrittenFilter.getPredicate().equals(oldPlan.getPredicate())) {
+        if (!ImmutableSet.copyOf(extractConjuncts(rewrittenFilter.getPredicate())).equals(ImmutableSet.copyOf(extractConjuncts(oldPlan.getPredicate())))) {
```

Comment:
I don't know. Probably it would not happen. But it would be better to be on the safe side. Use of `EE` seems good choice. I did not know about its existence :).